pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go

```diff
@@ -1089,7 +1090,7 @@ func checkNetworkStatusIPFamilies(ctx context.Context, o *operation.Operation) e
 		return err
 	}
 	condition := v1beta1helper.GetCondition(shoot.Status.Constraints, gardencorev1beta1.ShootToDualStackMigration)
-	if condition != nil {
+	if condition != nil && condition.Status == gardencorev1beta1.ConditionDualStackNodes {
```

Comment:
Should the state transitions be documented somewhere, e.g. at the condition type or in documentation?